### PR TITLE
refactor(path): prepare for `noUncheckedIndexedAccess`

### PR DESCRIPTION
**Pull request by javihernant**

### CODE REVIEW

File: path/_common/glob_to_reg_exp.ts [hunk 1]
```diff
@@ -65,7 +65,7 @@ export function _globToRegExp(
 
   // Remove trailing separators.
   let newLength = glob.length;
-  for (; newLength > 1 && c.seps.includes(glob[newLength - 1]); newLength--);
+  for (; newLength > 1 && c.seps.includes(glob[newLength - 1]!); newLength--);
```
Comment (javihernant) [inline path/_common/glob_to_reg_exp.ts:68]: `glob` is accessing their values using the `newLength` that is (initially) the length of `glob`. So, using `!` is justified as the elements accessed will always be defined

File: path/windows/to_file_url.ts [hunk 2]
```diff
@@ -20,10 +20,12 @@ export function toFileUrl(path: string): URL {
...
   }
-
   const [, hostname, pathname] = path.match(
     /^(?:[/\\]{2}([^/\\]+)(?=[/\\](?:[^/\\]|$)))?(.*)/,
   )!;
+  if (!pathname) {
+    throw new TypeError("Invalid pathname.");
+  }
   const url = new URL("file:///");
   url.pathname = encodeWhitespace(pathname.replace(/%/g, "%25"));
```
Comment (javihernant) [inline path/windows/to_file_url.ts:30]: to use `pathname.replace()` first it's needed to check that pathname is not `undefined` (due to maybe a failed regex match)

Comment (iuioiua) [reply path/windows/to_file_url.ts:30]: Are you able to add a test for this new case?

**Review by iuioiua — APPROVED** [review]
LGTM! Thank you.

**Comment by iuioiua** [comment]
Towards #4040

**Pull request merged**